src/_blog/js-libp2p-v1.md

```diff
@@ -0,0 +1,133 @@
+
+The circuit relay protocol was inspired by [TURN](https://en.wikipedia.org/wiki/Traversal_Using_Relays_around_NAT?useskin=vector), which is part of the Interactive Connectivity Establishment collection of NAT traversal techniques. Circuit Relay v2 has significant improvements over v1, including:
+
+- Support for resource reservation which allows for more efficient use of relay resources through explicit reservations.
+- Limitations placed durations and data caps on relayed connections.
+- Introduction of two subprotcols being `hop` and `stop` which were used to reserve and govern these resources effectively.
+
+You can read more about Circuit Relay v2 [here](https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md)
+
+
+## Webtransport
+
+js-libp2p is currently the only browser libp2p implementation that supports WebTransport. WebTransport is a new web standard that allows for the creation of bidirectional, multiplexed connections between a client and a server. WebTransport is a low-level API that can be used to implement higher-level protocols, such as QUIC. WebTransport is a major advantage for js-libp2p, and is a major advantage for the browser ecosystem as it allows for the creation of a low-latency, high-bandwidth connection between two peers. You can read more about WebTransport [here](https://github.com/libp2p/specs/tree/master/webtransport).
+
+
+# Performance Improvements and Optimizations 🔧
+
+We've made a variety of performance improvements and optimizations to js-libp2p based on feedback we've received from the community, as well as our performance benchmarking and profiling. We've removed a lot of async operations, fine tuned our streams, reduced round trip times and memory usage. Here's some of the major milestones we've achieved:
+
+## Perf Protocol
+
+The [perf protocol](https://github.com/libp2p/specs/blob/master/perf/perf.md) was created to help us analyze performance libp2p implementations through client-driven benchmarks. This revealed a variety of performance issues, and subsequently, allowed us to make a variety of optimizations. You can read more about the specifics of the test setup [here](https://github.com/libp2p/test-plans/tree/master/perf) but essentially two libp2p nodes are spun up on different AWS servers, and they then send data to each other continously over a period of time. The data is then analyzed to determine the performance of the libp2p implementation, you can view these visualizations on our [performance dashboard](https://observablehq.com/@libp2p-workspace/performance-dashboard). Here are some of the optimizations we made based on the results of the perf protocol:
+
+### Reducing latency
+On average 60% improvement in the time it takes to establish a connection between two nodes.
+
+<div class="container" style="display:flex; column-gap:10px; justify-content: center; align-items: center;">
+    <figure>
+        <img src="../assets/js-libp2p-v1-latency-improvements.png" width="750">
```

Comment:
I think latency is even better now, results can be rerun

Reply:
I thought latency should be around 0.3, see https://observablehq.com/@libp2p-workspace/performance-dashboard?branch=c8022cb77397759bd7e71a73e93f9074854989fe from https://github.com/libp2p/js-libp2p/pull/2253

Reply:
Looks like there are [other optimizations](https://github.com/libp2p/test-plans/tree/test/uint8arraylists-all-the-way-down) being made as well I will put this in draft until those are done.

Reply:
Latency & throughput are both improved since these were run but we need to update noise to depend on the 1.x release modules before we can run the benchmarks again.

Reply:
@achingbrain I saw that https://github.com/ChainSafe/js-libp2p-noise/pull/391 was merged and v14.0.0 was released, would you be able to re-run the benchmarks?